src/rust/engine/src/externs.rs

```diff
@@ -182,9 +178,11 @@ pub fn call(func: &Value, args: &[Value]) -> Result<Value, Failure> {
 /// NB: Panics on failure. Only recommended for use with built-in functions, such as
 /// those configured in types::Types.
 ///
-pub fn unsafe_call(func: &Function, args: &Vec<Value>) -> Value {
-  call(&val_for_id(func.0), args).unwrap_or_else(|e| {
-    panic!("Core function `{}` failed: {:?}", id_to_str(func.0), e);
+pub fn unsafe_call(func: &Function, args: &[Value]) -> Value {
+  let interns = INTERNS.read().unwrap();
```

Comment:
In order to avoid cloning the value, yea: it needs to be held. It's a tradeoff: `Value::clone()` isn't cheap, and so we're avoiding an extern call by holding a (shared) lock longer.